runtime/compiler/control/JITServerHelpers.cpp

```diff
@@ -1507,3 +1537,49 @@ JITServerHelpers::getFullClassName(uint8_t *name, uint32_t length, const J9ROMCl
    if (!primitive)
       name[i + baseNameLength] = ';';
    }
+
+
+size_t
+JITServerHelpers::getGeneratedClassNamePrefixLength(const J9UTF8 *name)
```

Comment:
The comment is above the function declaration in the .hpp file.

I don't think the VM has a concept of a "runtime-generated class" that matches what we need here. There might be ways to optimize the check for some of these cases, but I think it would better be done as a separate PR.

Reply:
I agree that right now the VM does not have what we need. I was merely suggesting to add such information.
A separate PR is the right way to do it.